src/clip.sol

```diff
@@ -252,46 +251,56 @@ contract Clipper {
                   address who,          // Receiver of collateral, payer of DAI, and external call address
                   bytes calldata data   // Data to pass in external call; if length 0, no call is done
     ) external lock isStopped(2) {
-        // Read auction data
-        Sale memory sale = sales[id];
-        require(sale.usr != address(0), "Clipper/not-running-auction");
+        address usr = sales[id].usr;
+        uint96  tic = sales[id].tic;
```

Comment:
Yes, but you need to load it together with `usr` in order to just use one `SLOAD`. If you separate them, the compiler will do 2.

Reply:
I see. Since `usr` is an address, i.e. 20 bytes, and `tic` is a uint96, i.e. 12 bytes, the total is 32 bytes, so I suppose `SLOAD`s are able to group together things that are 32 bytes long.

Reply:
Exactly. But it's because they are actually in the same storage slot and that depends in the order the storage variables are defined. The same for reading, in order the compiler takes advantage of this, they both need to be called together.